src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java

```diff
@@ -62,62 +71,66 @@ protected CompletableFuture<Void> processHoldings(OrderLineUpdateInstanceHolder
 
 
   private CompletableFuture<Void> moveHoldings(OrderLineUpdateInstanceHolder holder, String newInstanceId, RequestContext requestContext) {
-    List<String> holdingIds = holder.getStoragePoLine()
-      .getLocations()
-      .stream()
-      .map(Location::getHoldingId)
-      .filter(Objects::nonNull)
-      .collect(toList());
-    holdingIds.forEach(id -> holder.addHoldingRefsToStoragePatchOrderLineRequest(id, id));
-
-    return inventoryManager.getHoldingsByIds(holdingIds, requestContext)
-        .thenCompose(holdings -> inventoryManager.updateInstanceForHoldingRecords(holdings, newInstanceId, requestContext));
+    return pieceStorageService.getPiecesByPoLineId(PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine()), requestContext)
+      .thenApply(pieces -> {
+        List<String> holdingIds = Stream.concat(pieces.stream().map(Piece::getHoldingId),
```

Comment:
please add a small comment describing what the code in the `thenApply` is doing.

Reply:
Done